### DISCUSSION TITLE
rtl433 event in HomeAssistant -- how?

### DISCUSSION THREAD

**Discussion by eigenphase** [merbanan/rtl_433#2603] [Q&A] [answered]
Any MQTT/HomeAssistant users here? I am looking for a way to have an event in Home Assistant when a certain wireless signal is received. Starting from latest Home Assistant release, there is a promising [Event entity](https://www.home-assistant.io/integrations/event/), along with its [MQTT counterpart](https://www.home-assistant.io/integrations/event.mqtt/).

But I am not sure how to get this working and I did not find any infos on it.

In my case, I added my Reolink doorbell to rtl433 as described [here](https://github.com/merbanan/rtl_433/issues/2277). Whenever I push the button, an MQTT message is generated which always looks like this:

<img width="236" alt="image" src="https://github.com/merbanan/rtl_433/assets/7464149/d914dc0f-f1cd-4086-83b8-1cc2f3fb7626">

It has the usual time, mod, freq, rssi, snr, noise fields as well as some others like data and id which are fixed.

Now I have created the following topic for auto discovery with Home Assistant:

```
homeassistant/event/Reolink-6224189/config:
{
  "state_topic": "rtl_433/+/devices/Reolink-Doorbell/6224189/data",
     "event_types": [ "5ef93d8" ],
  "name": "Doorbell Button",
  "device_class": "doorbell",
  "unique_id": "rtl433_doorbell_6224189_event_button",
  "object_id": "rtl433_doorbell_6224189_event_button",
  "device": {
    "identifiers": "rtl433_doorbell_6224189",
    "name": "Reolink POE Doorbell Button",
    "model": "POE Doorbell",
    "manufacturer": "Reolink"
  }
}
```

Indeed, this now shows up as device in HomeAssistant and the Event entity is also there:

<img width="486" alt="image" src="https://github.com/merbanan/rtl_433/assets/7464149/489e00f6-961d-4473-a982-4a006edaab3b">

But when I push the button, nothing happens ... but the MQTT message is definitely generated by rtl433.

Any idea what's wrong here?
Is this even the right way?

If not, how do I otherwise get the doorbell button into Home Assistant?

**Comment by gdt** [comment]
I think you are having a HA MQTT autodiscovery problem, not an rtl-433 problem.   Within that, I think the problem is that while you created the event, you did not provide a mapping from any topic/json to the event, like you would have with autoconfguring a temperature sensor.

Can you explain the big picture of this doorbell?  It is said to be POE, which makes me think it is a camera also, and I would expect the button press to be available over the IP api.    But I think you are decoding a signal sent from the doorbell to a chime, similar to how non-IP regular wireless doorbells work.  Is that right?

**Reply by eigenphase** [reply]
Yes it’s POE but it also sends the signal over 433MHz (to its wireles chime as you say). I’d like to use that for reliability reasons rather than the API signal (and/or both).

My question  is really generic though: How do I bring an event in rtl433 (=certain message was received via 433MHz) into HA? Preferably with the new Event entity.

**Comment by gdt** [comment]
You  read the HA documentation about what kind of configuration is needed.  I am pretty sure that invovles configuring an mqtt event entity with a topic and a value_template, just like you would do for a sensor or switch, and that you need to add that to what you are doing.  if you are seeing the mqtt write when you push the button, the rtl_433 part is ok and your issue is just HA config.

**Comment by gdt** [comment, marked as answer]
See topic and value_template at https://www.home-assistant.io/integrations/event.mqtt

**Reply by eigenphase** [reply]
Re-emphasizing that I am aware this is not strictly rtl433 specific but I was still asking because I thought people here might know.
Indeed, you brought me to the right track -- thank you!

Solution: The MQTT Event expects a json object.

So I just changed the discovery config to:

```
{
  "state_topic": "rtl_433/+/devices/Reolink-Doorbell/6224189/data",
  "event_types": [ "press" ],
  "name": "Doorbell Button",
  "device_class": "doorbell",
  "unique_id": "rtl433_doorbell_6224189_event_button",
  "object_id": "rtl433_doorbell_6224189_event_button",
  "value_template": "{\"event_type\": \"press\"}",
  "device": {
    "identifiers": "rtl433_doorbell_6224189",
    "name": "Reolink POE Doorbell Button",
    "model": "POE Doorbell",
    "manufacturer": "Reolink"
  }
}
```

It basically ignores all the data.... it always sends data "{event_type: press}" via the value_template which is what is expected in "event_types". It always happens when the MQTT message is received, i.e., the doorbell button is actually pressed.
